ruby/ext/google/protobuf_c/map.c

```diff
@@ -572,6 +572,30 @@ static VALUE Map_freeze(VALUE _self) {
   return _self;
 }
 
+/*
+ * Deep freezes the map and values recursively.
+ * Internal use only.
+ */
+VALUE Map_internal_deep_freeze(VALUE _self) {
+  Map* self = ruby_to_Map(_self);
+
+  if (!RB_OBJ_FROZEN(_self)) {
```

Comment:
We probably should remove this `if`. There is no guarantee that our children are already frozen. And in the limited places where we are using this, we already know that it is not frozen.

Reply:
Ditto in the other places where we have `deep_freeze`